co-simulation/fed/mosaic-infrastructure/src/main/java/org/eclipse/mosaic/fed/infrastructure/ambassador/InfrastructureMessageAmbassador.java

```diff
@@ -149,44 +138,120 @@ private synchronized void receiveInteraction(InfrastructureV2xMessageReception i
+    }
 
+    /**
+     * Creates an Ad-Hoc configuration object to represent the configuration of the
+     * Ad-Hoc interface used for communication between the infrastructure instance
+     * and other vehicles or components, and sends it to the RTI for exchange.
+     *
+     * @param reg the infrastructure registration message received from the RTI
+     */
+    private final void onDsrcRegistrationRequest(InfrastructureRegistrationMessage reg) {
+        // Create an InterfaceConfiguration object to represent the configuration of the
+        // Ad-Hoc interface
+        // Set the IP address and subnet mask to null for now
+        // Set the transmit power to 50 dBm and the maximum range to 100 meters
+        InterfaceConfiguration interfaceConfig = new InterfaceConfiguration.Builder(AdHocChannel.SCH1)
+                .ip(null)
+                .subnet(null)
+                .power(50)
+                .radius(100.0)
+                .create();
+
+        // Create an AdHocConfiguration object to associate the Ad-Hoc interface
+        // configuration
+        // with the infrastructure instance's ID
+        AdHocConfiguration adHocConfig = new AdHocConfiguration.Builder(reg.getInfrastructureId())
+                .addInterface(interfaceConfig)
+                .create();
+
+        // Create an AdHocCommunicationConfiguration object to specify the time and
+        // Ad-Hoc
+        // configuration for exchange with another vehicle or component
+        AdHocCommunicationConfiguration communicationConfig = new AdHocCommunicationConfiguration(currentSimulationTime,
+                adHocConfig);
+
+        try {
+            // Trigger RTI interaction to MOSAIC to exchange the Ad-Hoc configuration
+            this.rti.triggerInteraction(communicationConfig);
+        } catch (InternalFederateException | IllegalValueException e) {
+            // Log error message if there was an issue with the RTI interaction
+            log.error(e.getMessage());
+        }
     }
 
     /**
-     * This method is called by the AbstractFederateAmbassador when a time advance
-     * has been granted by the RTI. Before this call is placed, any unprocessed
-     * interaction is forwarded to the federate using the processInteraction method.
+     * Performs registration of a new infrastructure instance and sets up
+     * communication interfaces.
      *
-     * @param time The timestamp towards which the federate can advance it local
-     *             time.
+     * @param reg The registration message of the new infrastructure instance.
+     */
+    private final void onRsuRegistrationRequest(InfrastructureRegistrationMessage reg) {
+
+        // Register the new infrastructure instance to the RTI as an RSU
+        RsuRegistration rsuRegistration = new RsuRegistration(currentSimulationTime, reg.getInfrastructureId(),
+                null, null,
```

Comment:
`null` might be a problem unless you've tested this already. I could see the MOSAIC RTI choking if passed null here, unless they're on top of their null-checks. I think if we really want to say nothing for the group we should just use the empty string "", and for the applications I'd go with an empty-list.